man/sd_uid_get_state.xml

```diff
@@ -126,16 +133,23 @@
     of the "primary" session of a user. If the user has graphical
     sessions, it will be the oldest graphical session. Otherwise, it
     will be the oldest open session.</para>
+
+    <para><function>sd_uid_get_login_time()</function> may be used to
+    determine the login time of the user identified by the specified
```

Comment:
hmm, i think we need to be more accurate here. "login time" is just too vague.

Maybe say "the time of the first of the user's current sessions".

And also say "This call will fail with <constant>ENXIO</constant> if the user is not currently logged in", i.e it's important to notice that this carries no historical information, but is purely about what's currently going on.

Oh, and we probably should say `<constant>CLOCK_REALTIME</constant> time in microseconds" somewhere. 

In fact we probably should mention `CLOCK_REALTIME` and "microseconds" for `sd_session_get_start_time()` too, now that i think of it.

Reply:
> Maybe say "the time of the first of the user's current sessions".

It's not the first current session, in this case I could use the `sd_session_get_start_time()` function.
The behavior for this example is:
* Login on tty1
* Login on tty2
* Logout from tty1

`sd_session_get_start_time()` will only report login time for tty2, but `sd_uid_get_login_time()` still shows the time of tty1, so the time the user logged in the first time without completely being logged out since then.
If somebody has a good description for that behavior I'm happy to adjust the documentation.

Reply:
right. "the time when the user first logged in and remained continuously logged in since with at least one session without interruption, even if the first session they logged in with has already been terminated"

or something like that.

yikes, this is hard.

Reply:
or alternatively.

"the time the user's service manager has been invoked, which is either the time when the user's first session since which they stayed logged in continuously began, or — in case lingering is enabled – shortly after boot."

Reply:
or some mixture of the two?

Reply:
Ok, the description is updated.